src/extensibility/node/ExtensionManagerDomain.js

```diff
@@ -97,17 +97,80 @@ function _performInstall(packagePath, installDirectory, validationResult, callba
         var sourceDir = path.join(validationResult.extractDir, validationResult.commonPrefix);
 
         fs.copy(sourceDir, installDirectory, function (err) {
-            if (err) {
+
+            var fail = function (err) {
                 _removeFailedInstallation(installDirectory);
                 callback(err, null);
-            } else {
+            };
+
+            var finish = function () {
                 // The status may have already been set previously (as in the
                 // DISABLED case.
                 if (!validationResult.installationStatus) {
                     validationResult.installationStatus = Statuses.INSTALLED;
                 }
                 callback(null, validationResult);
+            };
+
+            if (err) {
+                return fail(err);
             }
+
+            var packageJson;
+
+            try {
+                packageJson = JSON.parse(fs.readFileSync(path.join(installDirectory, "package.json"), {
+                    encoding: "utf8"
+                }));
+            } catch (e) {
+                packageJson = null;
+            }
+
+            if (packageJson && packageJson.dependencies) {
```

Comment:
Could you invert this condition and call immediatly `return finish();`?
We could reindent the block in this way and it would immediately clear we are looking for dependencies only (without looking at the else branch)